index.js

```diff
@@ -68,7 +68,27 @@ module.exports = function (str) {
   } else if ((match = str.match(new RegExp(regexps['re1'], 'i')))) {
     let days = match[1];
     let period = match[2];
-
+    if (period === 'year') {
+      let year = parseInt(new Date().getFullYear(), 10) - days;
+      if (year < 0) {
+        return 'Before B.C.';
+      } else {
+        return [true, new Date(new Date(new Date(new Date().setFullYear(year)).setMonth(new Date().getMonth())).setDate(new Date().getDate()))];
```

Comment:
You don't need to set current month. Date object returned by `new Date()` corresponds to present date.